### ISSUE TITLE
Weekly timeseries plot not showing some data, inserting others

### ISSUE THREAD

**Issue by sgwinder** [OutdoorRD/trails-viz#128]
Data points are being dropped if they are too near the beginning of the month. For example, at Lennox Creek (siteid 181), the data for May 3, 2020 is not being displayed. I think that this could be a side effect of the "month" being `4` (since `weekstart` was in April), but `d2p` being in May? It also appears that the dashboard is displaying dates as `d2p + 1`. I think we should use either `d2p` or `weekstart` directly. I have been creating `weekstart` to be the first date in the week we are summarizing (always a Thursday), and made `d2p` the following Saturday. Maybe the most straightforward would be to create the plot with `weekstart`.

![image](https://user-images.githubusercontent.com/46452196/97233512-51d23600-179c-11eb-9317-3f1381de70ff.png)

![image](https://user-images.githubusercontent.com/46452196/97233923-0bc9a200-179d-11eb-8b72-40d3114554b8.png)

And, other data points seem to come out of nowhere, near the beginning of the year. Again at Lennox Creek, there is an extra data point between December 29, 2019 and January 5, 2020. This is not in the `viz_model_www.csv` table. 

![image](https://user-images.githubusercontent.com/46452196/97233137-ad4ff400-179b-11eb-84c8-6b6b716cc3ca.png)

![image](https://user-images.githubusercontent.com/46452196/97233331-0750b980-179c-11eb-9a14-afd29a9f0a4a.png)

I have no idea where this one came from!

**Comment by aromatic-toast** [comment]
@sgwinder, 
Can you refresh me on what `d2p` represents? 
And also, which project contains `lennox creek`? I was looking for it under `Western Washington` but was unable to find the polygon.

**Comment by sgwinder** [comment]
`d2p` stands for, I believe, "date to plot". Right @woodsp? Originally, `d2p` was not actually based on the underlying dates going into the estimates, but rather started on January 1 and then just added 7 to go through the year. I wanted more precision when I was summarising estimates and social media data, so I created `weekstart` and have been using it in my code. I've been creating `d2p` to write out for compatibility with the dashboard (but see Issue #126 ). 

And `Lennox Creek` is in the [Middle Fork project](https://trailtrends.outdoorrd.org/dashboard/MBS_SARL), sorry for the confusion.

**Comment by sgwinder** [comment]
This is still a problem. I think the fix is probably to create the weekly time series based on the `d2p` column directly.

An example of weird behavior is here, at the Pilchuck Lookout Trail in the South Mountain Loop project.

![image](https://user-images.githubusercontent.com/46452196/146592621-677d8671-7c80-4905-853f-096e764b2be5.png)

**Comment by EmiliaH** [comment]
@sgwinder, is this still a relevant issue?

**Comment by sgwinder** [comment]
Yes, this is still a problematic bug on the dashboard. Would be great to add to the list of potential improvements for David to investigate.

**Comment by EmiliaH** [comment]
Hey @sgwinder, considering our remaining effort for this project, is this still high enough of a priority to keep on our list? Do you have a sense of the effort required?

**Comment by sgwinder** [comment]
I'd very much like to get this fixed if it is easy (which I suspect it might be).

@davidye007 , could you please spend ~30 min looking into this and see if you agree with my guesses above about what might be causing the issue? And then make an estimate of how much time it would take to fix? Thank you!

**Comment by davidye007** [comment]
hey @sgwinder, I investigated into this issue, and believe the problem lies in how the frontend is calculating dates for the time series chart. 

Currently, the frontend used the features `['week', 'month', 'year']` to calculate the dates of every Sunday.

https://github.com/OutdoorRD/trails-viz/blob/5036bc19021afb419561b6c7c2866e4965178f98/trails-viz-app/src/components/TimeSeries.vue#L61-L72
https://github.com/OutdoorRD/trails-viz/blob/5036bc19021afb419561b6c7c2866e4965178f98/trails-viz-app/src/components/TimeSeries.vue#L173

The problem is the features `['week', 'month', 'year']` are not tied to Sundays. Rather, they appear to be associated with `weekstart`. The weird data points occur when the `week` feature shows up `x` times for the same month, but the number of Sundays show up for `x+1` or `x-1` times for the same `month`.

For example, in January 2020, there are 4 Sundays, but there are 5 `week` rows associated with January 2020. There are 5 `week` rows because there are 5 Thursdays in January 2020.
```
trail,week,month,year,jjmm,flickr,ebird,alltrails,uselevel,days_closed,twitter,reveal,wta,jjmmlg,d2p,weekstart,model,datecreated
181,1,1,2020,22,0,0,0,H,0,0,,0,3.13549,2020-01-04,2020-01-02,MBS_model,20240903
181,2,1,2020,24,0,0,0,H,0,0,,0,3.21888,2020-01-11,2020-01-09,MBS_model,20240903
181,3,1,2020,32,0,0,0,H,0,0,,0,3.49651,2020-01-18,2020-01-16,MBS_model,20240903
181,4,1,2020,28,0,0,0,H,0,0,,0,3.3673,2020-01-25,2020-01-23,MBS_model,20240903
181,5,1,2020,28,0,0,0,H,0,0,,0,3.3673,2020-02-01,2020-01-30,MBS_model,20240903
```
This is why we get a weird data point at `2020-01-2` for `Lennox Creek`. It should have been `2020-02-2`. But since `week` shows up 5 times for January 2020, the timeseries mistakenly think the 5th Sunday is also part of January, when it should be part of February.

I think using either `d2p` or `weekstart` for dates in the frontend will fix the date issues. This fix will involve both frontend and backend changes, and my time guesstimate is around 5 hrs.

Here are two possible ideas to fix the date issue:
1. Use `weekstart` as dates. With this approach, we would also need to generate a `weekstart` feature for `viz_model_wwwir.csv` (infrared sensor data) because it needs to merge with `viz_model_www.csv` on a common key.

2. Using `d2p` as dates. With this approach we don't need to make changes to the datasets since both `viz_model_www.csv` and `viz_model_wwwir.csv` have `d2p`.

@sgwinder Please let me know if this diagnosis makes sense, and if you are leaning towards any one of the solutions.

**Comment by sgwinder** [comment]
Thanks @davidye007 ! That does make sense. And I can see how it matched older versions of `['d2p', 'week', 'month', 'year']`, which used to be calculated based on the beginning of the year as well. For example, my comment about `d2p` above:
> Originally, d2p was not actually based on the underlying dates going into the estimates, but rather started on January 1 and then just added 7 to go through the year.

My preference is to use `weekstart` as the dates. This is the date that actually matches with the beginning of the data that are being summarized (the Thursday of each Thur-Wed week). It's easy to write this out for the `viz_model_wwwir.csv` on datasets going forward - it will require minimal changes in [`viz-model/weeklycombined_2_wwwir.R`](https://github.com/OutdoorRD/viz-model/blob/2624be3e94cdbaf62c9113591ea1e7056a68b419/weeklycombined_2_wwwir.R#L25). 

It's a little bit more challenging to regenerate all of the existing `viz_model_wwwir.csv` files that are being displayed on the dashboard, but I think this should also be manageable. `weekstart` is simply the date that is two days before `d2p`. You could make a simple script to loop through the files, calculate `weekstart` (for example, if you use the `lubridate` package in `R` it's the reverse of [this line](https://github.com/OutdoorRD/viz-model/blob/2624be3e94cdbaf62c9113591ea1e7056a68b419/weeklycombined_2_wwwir.R#L21), and write them back out. At that point I think it would also make sense to remove `d2p` entirely from those files. 

If you agree this is a good solution, I can make an issue in `viz-model` to do the minor updates required there to write out `weekstart` instead of `d2p` on future versions of `*wwwir.csv` and `*www.csv`.

**Comment by davidye007** [comment]
@sgwinder I am totally onboard with your solution. And I agree with using `weekstart`, especially since it matches with the data being summarized.

For the existing `*wwwir.csv` data, I can follow your steps and write a script that calculates `weekstart` as two days before `d2p` and then remove the `d2p` column. @EmiliaH could you please grant me access to the [viz-model](https://github.com/OutdoorRD/viz-model/) repository so can I view the script Sama shared? Thanks a lot!

@sgwinder I think you can create an issue in `viz-model` for the minor updates. Thanks!

**Comment by EmiliaH** [comment]
Alright @davidye007, you should be all set!

**Comment by sgwinder** [comment]
Great, thanks @davidye007 and @EmiliaH ! The script is nothing amazing: the relevant line of code is just `mutate(d2p = weekstart + 2)`. The magic is really the `lubridate` package that knows how to add/subtract dates in a way that recognizes the calendar. 

Also as a side note: we will continue having `d2p` in the `mmm` and `mmmir` files for now - I think this is ok. For those files, which are monthly, the `d2p` is simply MM-15-YYYY (eg the middle of the month that the data row represents).

**Comment by EmiliaH** [comment]
This issue impacts issue #79 so we should fix it it in the same PR!
